src/fileio/datafile.cxx

```diff
@@ -539,6 +541,8 @@ bool Datafile::write(const char *format, ...) const {
   tmp.openw(filename);
   bool ret = tmp.write();
   tmp.close();
+
+  delete[] filename;
```

Comment:
Ah, good point. In that case, should the destructor also check `filename` before deleting it?